logstash-core/src/main/java/org/logstash/jackson/StreamReadConstraintsUtil.java

```diff
@@ -78,6 +86,8 @@ StreamReadConstraints get() {
         if (configuredStreamReadConstraints == null) {
             final StreamReadConstraints.Builder builder = StreamReadConstraints.defaults().rebuild();
 
+            // Apply the Jackson defaults first, then the overrides from config
+            JACKSON_DEFAULTS.forEach((override, value) -> override.applicator.apply(builder, value));
```

Comment:
My biggest worry with these kind of changes is how to educate users to keep settings up to date.

1) Users upgraded LS from the versions which `jvm.options` doesn't have Jackson default params to current change, we need to make sure to inform them `jvm.options` is not up to date. The situations,
  - users who jumped from 8.12.0-, the `jvm.options` doesn't have 
  - users upgraded LS multiple times (8.12.0- to 8.15.0 then 8.17.0), the `jvm.options` may not have jackson default params, depending on how (through package managers, or download tarball and replace) they upgrade.

2) and with 1) we also need to inform LS is applying default values or user defined values are active. I think we are [covering this](https://github.com/elastic/logstash/blob/main/logstash-core/src/main/java/org/logstash/jackson/StreamReadConstraintsUtil.java#L101)

Reply:
I was thinking this approach *removed* the need for user's to worry about this at all. In all the cases you mentioned they never cared about this setting and now they can continue to not care about it. They only have to think about it if they are in a situation where they explicitly need to tune it. But maybe i'm not quite understanding your concern?